src/migrator/utils.ts

```diff
@@ -23,12 +24,49 @@ export type CategorizedMigrations = {
 type SbvrUtils = typeof sbvrUtils;
 export type MigrationTuple = [string, Migration];
 export type MigrationFn = (tx: Tx, sbvrUtils: SbvrUtils) => Resolvable<void>;
-export type Migration = string | MigrationFn;
 export type RunnableMigrations = { [key: string]: Migration };
 export type Migrations = CategorizedMigrations | RunnableMigrations;
+export type AsyncMigrationFn = (
+	tx: Tx,
+	sbvrUtils: SbvrUtils,
+) => Resolvable<Result>;
+
+export type AsyncMigration = {
+	type?: 'async';
+	fn?: AsyncMigrationFn | undefined;
+	sql?: string | undefined;
+	delayMS?: number | undefined;
+	backoffDelayMS?: number | undefined;
+	errorThreshold?: number | undefined;
+};
+export function isAsyncMigration(
+	migration: AsyncMigration | RunnableMigrations,
+): migration is AsyncMigration {
+	return (
+		(migration as AsyncMigration).type !== undefined &&
+		(migration as AsyncMigration).type === 'async'
```

Comment:
The undefined check is redundant:
```suggestion
		(migration as AsyncMigration).type === 'async'
```

Reply:
Thx!